public/index.html

```diff
@@ -25,7 +25,7 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
 
-    <link rel="icon" href="../assets/GCI.png" type="image/icon type" />
+    <link  href="%PUBLIC_URL%/GCI.png"  />
```

Comment:
you have to add [`type` and `rel` to `<link>` tag](https://www.w3schools.com/tags/att_link_sizes.asp).
You can also rename `public/GCI.png` to `public/favicon.ico`, and the browser will take care of the rest.